broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/NewPartitionTransitionImplTest.java

```diff
@@ -152,6 +152,94 @@ void shouldAbortOngoingTransitionWhenNewTransitionIsRequested() {
     invocationRecorder.verify(mockStep2).transitionTo(mockContext, secondTerm, secondRole);
   }
 
+  @Test
+  // regression test for https://github.com/camunda-cloud/zeebe/issues/7873
+  void shouldNotStartMultipleTransitions() {
+    // given
+    final var firstStepFirstTransitionFuture = TEST_CONCURRENCY_CONTROL.<Void>createFuture();
+    final var firstStepSecondTransitionFuture = TEST_CONCURRENCY_CONTROL.<Void>createFuture();
+    final var firstStepThirdTransitionFuture = TEST_CONCURRENCY_CONTROL.<Void>createFuture();
+    when(mockStep1.transitionTo(any(), anyLong(), any()))
+        .thenReturn(firstStepFirstTransitionFuture)
+        .thenReturn(firstStepSecondTransitionFuture)
+        .thenReturn(firstStepThirdTransitionFuture);
+    when(mockStep1.prepareTransition(any(), anyLong(), any()))
+        .thenReturn(TEST_CONCURRENCY_CONTROL.completedFuture(null));
+
+    final var transition = new NewPartitionTransitionImpl(of(mockStep1), mockContext);
+    transition.setConcurrencyControl(TEST_CONCURRENCY_CONTROL);
+
+    final var firstTransitionFuture = transition.transitionTo(1, Role.FOLLOWER);
+
+    // when
+    transition.transitionTo(2, Role.LEADER);
+    transition.transitionTo(2, Role.FOLLOWER);
+
+    firstStepFirstTransitionFuture.complete(null);
+    await().until(firstTransitionFuture::isDone);
+
+    // then
+    final var inOrder = inOrder(mockStep1);
+
+    // first transition sequence
+    inOrder.verify(mockStep1).onNewRaftRole(mockContext, Role.FOLLOWER);
+    inOrder.verify(mockStep1).transitionTo(mockContext, 1, Role.FOLLOWER);
```

Comment:
Please also verify calls to `prepareTransition` which is where the old close logic gets executed. This is part of the cleanup.

Reply:
Question on this one: How problematic is it if we close the step here as term: 2 and Leader, but never install the step again because it was canceled and then install it as follower? It seems we make decision based on this call and the role etc?

For example in this test the verification which succeeds would be `
    inOrder.verify(mockStep1).prepareTransition(mockContext, 2L, RaftServer.Role.LEADER);`

Reply:
Is this a blocker @pihme and @deepthidevaki ? But i think this was already previous the case since you could cancel an ongoing transition and it some steps will not be installed in the role, where it has been notified before.

Reply:
> Question on this one: How problematic is it if we close the step here as term: 2 and Leader, but never install the step again because it was canceled and then install it as follower? It seems we make decision based on this call and the role etc?

This should be safe to do. See the logs that you posted above. Even if we skip transition, we still do the cleanup's if there is anything to be done. We also set the last transition to the skipped transition. So when the last transition happens, it still correctly identifies as it is transitioning from the skipped role and makes the decisions based on that. If it decides it doesn't have to re-install a step, but that step doesn't exists it will install it.